integrations/zoom/auth.go

```diff
@@ -2,12 +2,123 @@ package zoom
 
 import (
 	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
 
-	"golang.org/x/oauth2"
+	"go.autokitteh.dev/autokitteh/integrations/common"
+	"go.autokitteh.dev/autokitteh/sdk/sdktypes"
+)
+
+var (
+	accountID    = sdktypes.NewSymbol("private_account_id")
+	clientID     = sdktypes.NewSymbol("private_client_id")
+	clientSecret = sdktypes.NewSymbol("private_client_secret")
 )
 
 // serverToken retrieves a Server-to-Server (2-legged OAuth) token, using the connection's
 // internal app details (based on: https://developers.zoom.us/docs/internal-apps/s2s-oauth/).
-func serverToken(_ context.Context, _ *privateApp) (*oauth2.Token, error) {
-	return nil, nil // TODO(INT-247): Implement.
+func serverToken(ctx context.Context, app *privateApp) (string, error) {
+	data := url.Values{}
+	data.Set("grant_type", "account_credentials")
+	data.Set("account_id", app.AccountID)
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost,
+		"https://zoom.us/oauth/token", strings.NewReader(data.Encode()))
+	if err != nil {
+		return "", fmt.Errorf("failed to create token request: %v", err)
+	}
+
+	req.SetBasicAuth(app.ClientID, app.ClientSecret)
+	req.Header.Set("Content-Type", "application/x-www-form-urlencoded")
+
+	client := &http.Client{Timeout: 10 * time.Second}
+	resp, err := client.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("token request failed: %v", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		body, _ := io.ReadAll(resp.Body)
+		return "", fmt.Errorf("failed to get token: status %d: %s", resp.StatusCode, string(body))
+	}
+
+	var tokenData tokenResp
+
+	if err := json.NewDecoder(resp.Body).Decode(&tokenData); err != nil {
```

Comment:
Merely declaring an empty var shouldn't be a separate block of code
```suggestion
	var tokenData tokenResp
	if err := json.NewDecoder(resp.Body).Decode(&tokenData); err != nil {
```